src/platform/testing/e2e/cypress/support/form-tester/index.js

```diff
@@ -126,40 +126,66 @@ const addNewArrayItem = $form => {
 };
 
 /**
- * Run the page hook if the page has one, optionally automatically fill the page
- * if no hook ran, expand any accordions, and run an aXe check. Finally, if the
- * page was automatically filled, also automatically continue to the next page.
+ * Performs the following actions on a page:
+ * 1. Run an initial axe check.
+ * 2. Run the page hook if the page has one.
+ * 3. Autofill if no hook ran and if the page is not review or confirmation.
+ * 4. Expand any accordions and run the end-of-page aXe check.
+ * 5. Run the post hook.
  *
  * @param {string} pathname - The pathname of the page to run the page hook on.
- * @param {boolean} [autofill] - If true, and if no page hook ran, automatically
- *     fill the page. If false, don't fill the page, even if no page hook ran.
  */
-const performPageActions = (pathname, autofill = true) => {
-  cy.execHook(pathname).then(hookExecuted => {
-    if (!hookExecuted && autofill) cy.fillPage();
+const performPageActions = pathname => {
+  cy.axeCheck();
+
+  cy.execHook(pathname).then(({ hookExecuted, postHook }) => {
+    const shouldAutofill = !pathname.match(
+      /\/(introduction|confirmation|review-and-submit)$/,
+    );
+
+    if (!hookExecuted && shouldAutofill) cy.fillPage();
 
     cy.expandAccordions();
     cy.axeCheck();
 
-    if (!hookExecuted && autofill) {
-      cy.findByText(/continue/i, { selector: 'button' }).click(FORCE_OPTION);
-    }
+    const postHookPromise = new Promise(resolve => {
+      postHook();
+      resolve();
+    });
+
+    cy.wrap(postHookPromise, NO_LOG_OPTION);
   });
 };
 
 /**
  * Top level loop that invokes all of the processing for a form page and
- * proceeds to the next page. When it gets to the end, it submits the form.
+ * asserts that it proceeds to the next page until it gets to the confirmation.
  */
 const processPage = () => {
-  // Run aXe check before doing anything on the page.
-  cy.axeCheck();
-
   cy.location('pathname', NO_LOG_OPTION).then(pathname => {
-    if (pathname.endsWith('review-and-submit')) {
-      performPageActions(pathname, false);
+    performPageActions(pathname);
 
-      // Check the privacy agreement box if it exists.
+    if (!pathname.endsWith('/confirmation')) {
+      cy.location('pathname', NO_LOG_OPTION)
```

Comment:
It's possible the submission timeout could still happen, but it's rare. Going to keep on eye on tests and re-introduce the extended timeout if it re-appears.